rocketpy/simulation/monte_carlo.py

```diff
@@ -181,186 +220,406 @@ def simulate(
 
         Returns
         -------
         None
         """
-        input_file.close()
-        output_file.close()
-        error_file.close()
+        sim_monitor = _SimMonitor(
+            initial_count=self._initial_sim_idx,
+            n_simulations=self.number_of_simulations,
+            start_time=time(),
+        )
+        try:
+            while sim_monitor.keep_simulating():
+                sim_monitor.increment()
+
+                inputs_dict, outputs_dict = MonteCarlo.__run_single_simulation(
+                    sim_monitor.count,
+                    self.environment,
+                    self.rocket,
+                    self.flight,
+                    self.export_list,
+                    self.export_sample_time,
+                )
+
+                MonteCarlo.__export_flight_data(
+                    inputs_dict,
+                    outputs_dict,
+                    self._input_file,
+                    self._output_file,
+                )
+
+                sim_monitor.print_update_status(sim_monitor.count)
+
+            sim_monitor.print_final_status()
 
-    def __terminate_simulation(self, input_file, output_file, error_file):
+        except KeyboardInterrupt:
+            MonteCarlo._reprint("Keyboard Interrupt, files saved.")
+            with open(self._error_file, "a", encoding="utf-8") as file:
+                file.write(json.dumps(inputs_dict, cls=RocketPyEncoder) + "\n")
+
+        except Exception as error:
+            MonteCarlo._reprint(
+                f"Error on iteration {self.__sim_monitor.count}: {error}"
+            )
+            with open(self._error_file, "a", encoding="utf-8") as file:
+                file.write(json.dumps(inputs_dict, cls=RocketPyEncoder) + "\n")
+            raise error
+
+    def __run_in_parallel(self, n_workers=None):
         """
-        Terminates the simulation, closes the files and prints the results.
+        Runs the monte carlo simulation in parallel.
 
         Parameters
         ----------
-        input_file : str
-            The file object to write the inputs.
-        output_file : str
-            The file object to write the outputs.
-        error_file : str
-            The file object to write the errors.
+        n_workers: int, optional
+            Number of workers to be used. If None, the number of workers
+            will be equal to the number of CPUs available. Default is None.
 
         Returns
         -------
         None
         """
-        final_string = (
-            f"Completed {self.__iteration_count} iterations. Total CPU time: "
-            f"{process_time() - self.__start_cpu_time:.1f} s. Total wall time: "
-            f"{time() - self.__start_time:.1f} s\n"
+        if n_workers is None or n_workers > os.cpu_count():
+            n_workers = os.cpu_count()
+
+        if n_workers < 3:
+            raise ValueError("Number of workers must be at least 3 for parallel mode.")
+
```

Comment:
I changed it to 2, I tried with only one worker, but it does not work since there is a minimum of two processes: the sim producer and the sim exporter.